components/formats-gpl/src/loci/formats/in/ColumbusReader.java

```diff
@@ -0,0 +1,737 @@
+          p.file = new Location(parent, value).getAbsolutePath();
+
+          String buffer = attrs.getNamedItem("BufferNo").getNodeValue();
+          p.fileIndex = Integer.parseInt(buffer);
+        }
+        else if (name.equals("Row")) {
+          p.row = Integer.parseInt(value) - 1;
+        }
+        else if (name.equals("Col")) {
+          p.col = Integer.parseInt(value) - 1;
+        }
+        else if (name.equals("FieldID")) {
+          p.field = Integer.parseInt(value) - 1;
+        }
+        else if (name.equals("TimepointID")) {
+          p.timepoint = Integer.parseInt(value) - 1;
+          if (p.timepoint == 0) {
+            p.timepoint = externalTime;
+          }
+        }
+        else if (name.equals("ChannelID")) {
+          p.channel = Integer.parseInt(value) - 1;
+        }
+        else if (name.equals("ChannelName")) {
+          p.channelName = value;
+        }
+        else if (name.equals("ChannelColor")) {
+          Long color = new Long(value);
+          int blue = (int) ((color >> 24) & 0xff);
+          int green = (int) ((color >> 16) & 0xff);
+          int red = (int) ((color >> 8) & 0xff);
+          int alpha = (int) (color & 0xff);
+
+          // not setting this yet, and deferring to the
+          // emission wavelength to set the color
+          // neither BGRA nor ARGB seems to make sense for all channels
+          //p.channelColor = new Color(red, green, blue, alpha);
+        }
+        else if (name.equals("MeasurementTimeOffset")) {
+          p.deltaT = new Double(value);
+        }
+        else if (name.equals("AbsTime")) {
+          p.deltaT = new Timestamp(value).asInstant().getMillis() / 1000d;
+        }
+        else if (name.equals("MainEmissionWavelength")) {
+          p.emWavelength = new Double(value);
+        }
+        else if (name.equals("MainExcitationWavelength")) {
+          p.exWavelength = new Double(value);
+        }
+        else if (name.equals("ImageResolutionX")) {
+          String unit = attrs.getNamedItem("Unit").getNodeValue();
+          p.sizeX = correctUnits(new Double(value), unit);
+        }
+        else if (name.equals("ImageResolutionY")) {
+          String unit = attrs.getNamedItem("Unit").getNodeValue();
+          p.sizeY = correctUnits(new Double(value), unit);
+        }
+        else if (name.equals("PositionX")) {
+          String unit = attrs.getNamedItem("Unit").getNodeValue();
+          p.positionX = correctUnits(new Double(value), unit);
+        }
+        else if (name.equals("PositionY")) {
+          String unit = attrs.getNamedItem("Unit").getNodeValue();
+          p.positionY = correctUnits(new Double(value), unit);
+        }
+        else if (name.equals("PositionZ")) {
+          String unit = attrs.getNamedItem("Unit").getNodeValue();
+          p.positionZ = correctUnits(new Double(value), unit);
+        }
+      }
+
+      planes.add(p);
+    }
+
+  }
+
+  private Double correctUnits(Double v, String unit) {
```

Comment:
Had a look at the data the metadata is stored in meters, meaning the values are low enough that they are below `Constants.EPSILON` and would be converted anyways. Given the late hour and the low impact, it might be easier to keep the implementation as it is.